client/src/pages/private/PSIView.js

```diff
@@ -108,14 +108,15 @@ export default () => {
         const data = await response.json();
         const mappedData = data.map((row) => {
           const rowCohorts = cohorts.filter((cohort) => cohort.psi_id === row.id);
+          // To calculate available_seats, we filter out the expired cohorts and
+          // then sum the remaining seats of the current ones
           return {
             ...row,
             id: row.id,
             cohorts: rowCohorts.length,
-            available_seats: rowCohorts.reduce(
-              (sum, cohort) => sum + (cohort.cohort_size - cohort.participants.length),
-              0
-            ),
+            available_seats: rowCohorts
```

Comment:
To improve efficiency we can declare new Date() in top scope. So, that help us to avoid every time creation new Date(). Just a thought sharing with you.